### DISCUSSION TITLE
React.ElementType issues

### DISCUSSION THREAD

**Discussion by rpcomprend** [Sitecore/jss#1765] [Help]
### Description

React.ElementType relies on JSX.IntrinsicElements, which is augmented with additional components such as 'feaas-picker'. For example, following definition:
```typescript
type PolymorphicRef<T extends React.ElementType> =
  React.ComponentPropsWithRef<T>['ref'];
```
results in type error. This complication mirrors experiences encountered when integrating with libraries like Chakra.

Currently, we're resorting to substituting React.ElementType with a custom type that leverages the native JSX.IntrinsicElements. However, this workaround isn't optimal.

Are there any suggestions for improvement?

**Comment by art-alexeyenko** [comment]
Hi @rpcomprend ! Could you please describe your usecase and the conditions where you encounter the error in more detail please? I think JSS does not use feaas-picker component directly.

**Reply by rpcomprend** [reply]
Thanks for reaching out! Happy to give you more info.

I created a Next.js project using `pnpm dlx create-sitecore-jss`:
```bash
pnpm list "*sitecore*" "@types/react"

dependencies:
@sitecore-jss/sitecore-jss-nextjs 21.7.1

devDependencies:
@sitecore-jss/sitecore-jss-cli 21.7.1
@sitecore-jss/sitecore-jss-dev-tools 21.7.1
@types/react 18.2.75
```

In any `.tsx` file I get
![obraz](https://github.com/Sitecore/jss/assets/160010827/04719413-e959-4903-add6-e55c99b00377)

The goal is to have polymorphic components, but it's not possible with feaas-* and byoc-registration elements in IntrinsicElements scope.
